app/src/main/java/org/mozilla/fenix/settings/logins/controller/SavedLoginsStorageController.kt

```diff
@@ -0,0 +1,200 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.logins.controller
+
+import android.content.Context
+import android.util.Log
+import androidx.navigation.NavController
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import mozilla.components.concept.storage.Login
+import mozilla.components.service.sync.logins.InvalidRecordException
+import mozilla.components.service.sync.logins.LoginsStorageException
+import mozilla.components.service.sync.logins.NoSuchRecordException
+import org.mozilla.fenix.HomeActivity
+import org.mozilla.fenix.R
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.settings.logins.LoginsAction
+import org.mozilla.fenix.settings.logins.LoginsFragmentStore
+import org.mozilla.fenix.settings.logins.fragment.EditLoginFragmentDirections
+import org.mozilla.fenix.settings.logins.mapToSavedLogin
+
+/**
+ * Controller for all saved logins interactions with the password storage component
+ */
+open class SavedLoginsStorageController(
+    context: Context,
+    private val viewLifecycleScope: CoroutineScope,
+    private val navController: NavController,
+    private val loginsFragmentStore: LoginsFragmentStore
+) {
+    private val activity: HomeActivity = context as HomeActivity
+
+    private suspend fun getLogin(loginId: String): Login? =
+        activity.components.core.passwordsStorage.get(loginId)
+
+    fun delete(loginId: String) {
```

Comment:
Could we give these slightly more descriptive names for easier parsing elsewhere?